app/src/main/java/dadeindustries/game/gc/ai/Mind.java

```diff
@@ -11,49 +20,285 @@
+		}
+		return enemyShips;
+	}
+
+	/**
+	 * Returns a list of all sectors in the galaxy that have systems that are inhabited by enemies.
+	 *
+	 * @param globalGameData
+	 * @return List of Sectors that are inhabited by the enemy
+	 */
+	private List<Sector> getInhabitedEnemySectors(GlobalGameData globalGameData) {
+		List<Sector> sectorsWithInhabitedEnemySystemsIn = new LinkedList<Sector>();
+		for (Sector sector : getSectorsWithSystemsIn(globalGameData)) {
+			if (sector.getSystem().hasFaction() && sector.getSystem().getFaction() != getPlayer().getFaction()) {
+				sectorsWithInhabitedEnemySystemsIn.add(sector);
+			}
+		}
+		return sectorsWithInhabitedEnemySystemsIn;
+	}
+
+	/**
+	 * Returns a list of all sectors in the galaxy that have systems that are uninhabited.
+	 *
+	 * @param globalGameData
+	 * @return List of Systems that are uninhabited
+	 */
+	private List<Sector> getSectorsWithUninhabitedSystems(GlobalGameData globalGameData) {
+		List<Sector> sectorsWithUninhabitedSystemsIn = new LinkedList<Sector>();
+		for (Sector sector : getSectorsWithSystemsIn(globalGameData)) {
+			if (!sector.getSystem().hasFaction()) {
+				sectorsWithUninhabitedSystemsIn.add(sector);
+			}
+		}
+		return sectorsWithUninhabitedSystemsIn;
+	}
+
+	private List<Sector> getSectorsWithSystemsIn(GlobalGameData globalGameData) {
+		List<Sector> sectorsWithSystemsIn = new LinkedList<Sector>();
+		Sector[][] galaxy = globalGameData.getSectors();
+		for (Sector[] sectors : galaxy) {
+			for (Sector sector : sectors) {
+				if (sector.hasSystem()) {
+					sectorsWithSystemsIn.add(sector);
+				}
+			}
+		}
+		return sectorsWithSystemsIn;
+	}
+
+	private List<ColonyShip> getColonyShips(List<Spaceship> spaceshipList) {
+		List<ColonyShip> colonyShips = new LinkedList<ColonyShip>();
+		for (Spaceship ship : spaceshipList) {
+			if (ship instanceof ColonyShip) {
+				colonyShips.add((ColonyShip) ship);
+			}
+		}
+		return colonyShips;
+	}
+
+	private List<CombatShip> getCombatShips(List<Spaceship> spaceshipList) {
+		List<CombatShip> combatShips = new LinkedList<CombatShip>();
+		for (Spaceship ship : spaceshipList) {
+			if (ship instanceof CombatShip) {
+				combatShips.add((CombatShip) ship);
+			}
+		}
+		return combatShips;
+	}
+
+	private void buildNewShips(GlobalGameData globalGameData, List<System> systemList, List<Spaceship> spaceshipList) {
+		// Determine spaceship ratio
+		boolean shouldBuildCombatShips = shouldIBuildCombatShips(spaceshipList);
+		// If any system has empty build queue
+		for (System system : systemList) {
+			if (system.isBuildQueueEmpty()) {
+				addOptimalShipToBuildQueue(globalGameData, shouldBuildCombatShips, system);
+			}
+		}
+	}
+
+	private void addOptimalShipToBuildQueue(GlobalGameData globalGameData, boolean shouldBuildCombatShips, System system) {
+		Sector sectorSystemIsIn = globalGameData.getSectors()[system.getX()][system.getY()];
+		// Order any systems not already building a spaceship to build one of the most important type
```

Comment:
Which is the most of important type?

Reply:
The most important type is determined by a preset ratio of combat:colony ships. 
Currently, the ratio is set to 0.8, so 4 combat ships for every 1 colony ship. Whichever ship would bring the AIs forces closer to that ratio is the one it will build.